### PR TITLE
[Journal] Markdown to Formatted HTML Functionality

### LINKED ISSUE #69: [Journal] Markdown to Formatted HTML Functionality
### Objectives
The main objective for this issue is for the journal to be able to convert markdown to displayed formatted html. The user should be able to go into an "edit" mode and then they are able to input markdown syntax that is previewed/rendered live in formatted html. For example, if the user wants to bold using `**this**`, the previewed formatted html should be **this** (which is actually bolded).

### Acceptance Criteria
- [x] Implement/incorporate function to convert Markdown to Formatted HTML
- [x] The journal content is saved for each journal into localStorage
- [x] The journal content is fetched for each journal from localStorage and displayed on page
- [ ] Update E2E testing so pipeline does not fail with new changes

### Assignees
- @kennethkietvuong

### PR DESCRIPTION
**Pull request by kennethkietvuong**
# Objective
Refer to the corresponding GitHub issue #69.

The purpose of this PR is to implement and integrate the conversion of **markdown syntax into formatted html**, meaning when you use markdown syntax, the syntax is displayed visibly (such that when you bold text in markdown, it appears to be bold). Due to this change, the E2E tests were also updated accordingly to fit the new criteria.

Additionally, this PR refactors/cleans up the journal files a little bit by adding JSDoc function comments and removes some commented out code.

[demo.webm](https://github.com/cse110-sp24-group30/cse110-sp24-group30/assets/103289124/03e539c3-bdd6-49db-bcb6-67b17aa1d9f8)

## Solution

- Add JSDoc function comments
- Reworked journal functions to support markdown syntax to formatted html
  - Updated `journal-app.js`
  - Updated `journal-style.css`
  - Updated `journal-page.html`
- Updated E2E journal testing (`journal-add.test.js` → `journal.test.js`)
  - A single file to contain all journal tests

### Assignees
- @kennethkietvuong 

### Additional Notes
The E2E journal test file is not complete. I left comments at the bottom (in the test file) for what other instances need to be tested with the journal module.

### CODE REVIEW

**Comment by kabirsachdev7** [comment]
Need to check whether the PR is going to conflict with PR 51 as both are working on functionalities of the journal.

**Review by kabirsachdev7 — COMMENTED** [review]
Need to review whether this PR will conflict with PR 51 since they are both working on the functionalities of journal.

File: journal/journal-app.js [hunk 1]
```diff
@@ -20,99 +21,175 @@ function init() {
     addJournalNew(journalContainer, true);
 }
 
-// Display the new journal element to the page
+/**
+ * Creates and displays journal(s) onto the page
+ * 
+ * @param {HTMLElement} journalContainer - DOM element to display journals in
+ * @param {Boolean} existing - Displays existing journal(s) from localStorage if true
+ */
```
Comment (adironene) [inline journal/journal-app.js:29]: good job adding jsdoc!

File: journal/journal-style.css [hunk 2]
```diff
@@ -107,32 +134,35 @@ body {
     background-color: rgba(0, 0, 0, 0.5);
 }
 
+/* Container for edit/delete buttons */
 .button-container {
     position: absolute;
-    top: 5px;
-    right: 5px;
+    top: 200px;
```
Comment (adironene) [inline journal/journal-style.css:140]: would the rigid padding be a problem when the screen dimensions change?

**Review by adironene — COMMENTED** [review]
Looks really good! Good work guys.

Just left a few comments about some minor things. Also, demo works great, thanks for including that

File: __tests__/journal.test.js [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+
+        const htmlOutputHTML = await page.$eval('.journal-htmlOutput', el => el.innerHTML.trim());
+        expect(htmlOutputHTML).toBe('<h1>Hello</h1>\n<p><strong>Bold</strong></p>\n<p><em>Italics</em></p>\n<blockquote>\n<p>Block Quotes</p>\n</blockquote>\n<ol>\n<li>First Item</li>\n<li>Second Item</li>\n<li>Third Item</li>\n</ol>\n<ul>\n<li>First Item</li>\n<li>Second Item</li>\n<li>Third Item</li>\n</ul>\n<p><code>code</code></p>\n<hr>\n<p><a href="">Link</a></p>\n<p><img src="" alt="Alt Image"></p>');
+    }, 15000);
+
+    it('Saving a single journal w/ various markdown text', async () => {
+        console.log('Saving journal...');
+        await page.click('.save-button');
+
+        // Check header title
```
Comment (adironene) [inline __tests__/journal.test.js]: is this todo? if so, please add keyword

**Review by AliMohammadiasl — APPROVED** [review]
Looks really good! Really like how the first line counts as the title of the journal. Only comment I have is maybe we need to somehow tell the user that the first line in the journal page will be the title of the journal.

**Pull request merged**